### PR TITLE
Refactor time periods from flights. Add time periods to clock.

### PR DESCRIPTION
**Pull request by pmtischler**
For #45

### CODE REVIEW

File: server/auvsi_suas/models/time_period.py [hunk 1]
```diff
@@ -4,10 +4,50 @@ class TimePeriod(object):
...
+                of a period.
+            is_end_func: Lambda which returns whether an event is the end
+                of a period.
+        Returns:
+            A list of time periods.
+        """
+        periods = []
+
+        # Handle end with no start, use None as infinity.
+        if events and is_end_func(events[0]):
```
Comment (prattmic) [inline server/auvsi_suas/models/time_period.py]: Before this return [] if len(events) < 1

File: server/auvsi_suas/models/time_period.py [hunk 2]
```diff
@@ -4,10 +4,50 @@ class TimePeriod(object):
...
+
+        return periods
+
     def __init__(self, start=None, end=None):
         self.start = start
         self.end = end
 
+    def __unicode__(self):
+        """Descriptive text for use in displays."""
+        return unicode('TimePeriod (start: %s, end: %s)' % (self.start,
```
Comment (prattmic) [inline server/auvsi_suas/models/time_period.py]: Nit: `TimePeriod(start=%s, end=%s)`

File: server/auvsi_suas/models/time_period_test.py [hunk 3]
```diff
@@ -7,6 +8,62 @@
 class TestTimePeriod(TestCase):
     """Tests the TimePeriod object."""
 
+    def setUp(self):
+        self.Event = namedtuple('Event', ['timestamp', 'start'])
```
Comment (prattmic) [inline server/auvsi_suas/models/time_period_test.py]: Why make this a class attribute instead of just a normal global class?

**Comment by prattmic** [comment]
LGTM

File: server/auvsi_suas/models/time_period.py (hunk 1 above)
Comment (pmtischler) [inline server/auvsi_suas/models/time_period.py]: Done.
```
if not events:
  return []
```

File: server/auvsi_suas/models/time_period.py (hunk 2 above)
Comment (pmtischler) [inline server/auvsi_suas/models/time_period.py]: Done.

File: server/auvsi_suas/models/time_period_test.py (hunk 3 above)
Comment (pmtischler) [inline server/auvsi_suas/models/time_period_test.py]: Done.

**Pull request merged**